pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -291,7 +291,11 @@ func (lb *ListenerBuilder) buildSidecarOutboundListeners(node *model.Proxy,
 ) []*listener.Listener {
 	noneMode := node.GetInterceptionMode() == model.InterceptionNone
 
-	actualWildcard, actualLocalHostAddress := getActualWildcardAndLocalHost(node)
+	actualWildcardIPv4, actualWildcardIPv6, actualLocalHostAddrIPv4, actualLocalHostAddrIPv6, isDualStack := getDualStackWildcardAndLocalHost(node)
```

Comment:
My concerns around the complexity of this function are still present. Please abstract this into a struct or other structured form; a 5 tuple respones is almost impossible to use correctly and adds tons of complexity to all usages (as can be seen at each callsite in this PR)

Reply:
okay, sure, I will abstract it into a struct, thanks a lot for your comments @howardjohn!

Reply:
And I may add this into Proxy struct.